packages/twenty-server/src/modules/messaging/common/services/messaging-channel-sync-status.service.ts

```diff
@@ -160,5 +168,55 @@ export class MessagingChannelSyncStatusService {
       MessageChannelSyncStatus.FAILED_INSUFFICIENT_PERMISSIONS,
       workspaceId,
     );
+
+    await this.addToAccountsToReconnect(messageChannelId, workspaceId);
+  }
+
+  private async addToAccountsToReconnect(
```

Comment:
looks like to be duplicated logic. I would put it in a connectedAccount service (we need to double check the dependencies) that would take { workspaceId, connectedAccountId } as parameters